File: android/measure/src/main/java/sh/measure/android/tracing/InvalidSpan.kt
package sh.measure.android.tracing

import sh.measure.android.attributes.AttributeValue

internal class InvalidSpan : Span {
    override val traceId: String = "invalid-trace-id"
    override val spanId: String = "invalid-span-id"
    override val isSampled: Boolean = false
    override val parentId: String? = null

    override fun setStatus(status: SpanStatus): Span = this

    override fun setParent(parentSpan: Span): Span = this

    override fun setCheckpoint(name: String): Span = this

    override fun setName(name: String): Span = this

    override fun setAttribute(key: String, value: String): Span = this

    override fun setAttribute(key: String, value: Long): Span = this

    override fun setAttribute(key: String, value: Int): Span = this

    override fun setAttribute(key: String, value: Double): Span = this

    override fun setAttribute(key: String, value: Boolean): Span = this

    override fun setAttributes(attributes: Map<String, AttributeValue>): Span = this

    override fun removeAttribute(key: String): Span = this

    override fun end(): Span = this

    override fun end(timestamp: Long): Span = this

    override fun hasEnded(): Boolean = false

    override fun getDuration(): Long = 0
}

File: flutter/packages/measure_flutter/lib/src/tracing/invalid_span.dart
import 'package:measure_flutter/src/tracing/span.dart';
import 'package:measure_flutter/src/tracing/span_status.dart';

import '../attribute_value.dart';

class InvalidSpan implements Span {
  @override
  String get traceId => "invalid-trace-id";

  @override
  String get spanId => "invalid-span-id";

  @override
  bool get isSampled => false;

  @override
  String? get parentId => null;

  @override
  Span setStatus(SpanStatus status) {
    return this;
  }

  @override
  Span setParent(Span parentSpan) {
    return this;
  }

  @override
  Span setCheckpoint(String name) {
    return this;
  }

  @override
  Span setName(String name) {
    return this;
  }

  @override
  Span setAttributeString(String key, String value) {
    return this;
  }

  @override
  Span setAttributeInt(String key, int value) {
    return this;
  }

  @override
  Span setAttributeDouble(String key, double value) {
    return this;
  }

  @override
  Span setAttributeBool(String key, bool value) {
    return this;
  }

  @override
  Span setAttributes(Map<String, AttributeValue> attributes) {
    return this;
  }

  @override
  Span removeAttribute(String key) {
    return this;
  }

  @override
  Span end() {
    return this;
  }

  @override
  Span endWithTimestamp(int timestamp) {
    return this;
  }

  @override
  bool hasEnded() {
    return false;
  }

  @override
  int getDuration() {
    return 0;
  }
}
